common/persistence/sql/storage/store.go

```diff
@@ -67,7 +79,83 @@ func NewSQLDB(cfg *config.SQL) (sqldb.Interface, error) {
 	}
 	// Maps struct names in CamelCase to snake without need for db struct tags.
 	db.MapperFunc(strcase.ToSnake)
-	return mysql.NewDB(db, nil), nil
+	return db, nil
+}
+
+func RegisterTLSConfig(cfg *config.SQL) error {
+	host, _, err := getHostPortFromConnectAddr(cfg.ConnectAddr)
+	if err != nil {
+		log.Panicf("error getting host port from ConnectAddr: %v", err)
```

Comment:
Return an error from here and let the upper layer decide what to do with the error